src/main/java/com/faforever/moderatorclient/ui/main_window/LadderMapPoolController.java

```diff
@@ -57,8 +67,16 @@
     public ComboBox<MatchmakerQueue> queueComboBox;
     public ScrollPane bracketsScrollPane;
     public VBox addButtonsContainer;
+    public ComboBox<ComparableVersion> neroxisVersionComboBox;
+    public Spinner<String> neroxisSizeSpinner;
+    public Spinner<Integer> neroxisSpawnsSpinner;
 
-    private ObjectProperty<MapVersionFX> selectedMap = new SimpleObjectProperty<>();
+    private final ObjectProperty<MapPoolAssignmentFX> selectedMap = new SimpleObjectProperty<>();
+    private final BiPredicate<MapPoolAssignmentFX, MapPoolAssignmentFX> matchingPoolAssignmentPredicate = (assignmentFX1, assignmentFX2) ->
+            (assignmentFX1.getMapVersion() != null && assignmentFX1.getMapVersion().equals(assignmentFX2.getMapVersion()))
+                    || (assignmentFX1.getMapParams() != null && assignmentFX1.getMapParams().equals(assignmentFX2.getMapParams()));
```

Comment:
Use Objects.equals instead. Null safe and more readable

Reply:
Unfortunately can't use objects.equals as Objects.equals(null,null) = true which would be true for everything as either mapParams or mapVersion is null on each assignment